### PR TITLE
@alloy => Make a connection for partner artists (+ cache headers for unauth'ed requests)

### PR DESCRIPTION
**Pull request by mzikherman**
~~This doesn't actually work (yet), but uses the loaders in the way you might expect (ie- unauthenticated loaders can still cache and resolve with body/headers).~~

This PR does all the things now- it can cache/resolve body/headers for unauthenticated loaders, and also exposes an edge from artist to partners, decorated with `PartnerArtist` info.

### CODE REVIEW

**Comment by mzikherman** [comment]
cc @joeyAghion this is the partner artist connection that launched the caching unauthenticated headers discussion. Will try to add that as a commit here.

File: schema/partner_artist.js [hunk 1]
```diff
@@ -61,3 +62,7 @@ const PartnerArtist = {
 }
 
 export default PartnerArtist
+
+export const partnerArtistConnection = connectionDefinitions({
+  nodeType: PartnerArtistType,
```
Comment (alloy) [inline schema/partner_artist.js]: Wasn’t the original goal to make ‘partner artist’ an edge, rather than a node?

Comment (mzikherman) [reply schema/partner_artist.js]: Ah yes, good point!

**Comment by mzikherman** [comment]
Pushed a WIP commit that tries to create a connection using the built-in lib, and then trying to coerce more data into `edges`. Didn't....get very far.

**Comment by alloy** [comment]
I think fully handrolling the connection and edge types is the way to go

File: lib/loaders/api/loader_without_authentication_factory.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // @ts-check
```
Comment (mzikherman) [inline lib/loaders/api/loader_without_authentication_factory.js]: cc @joeyAghion this is what we were initially talking about, you can now cache body and headers for unauthenticated requests if the loader is specified with `header: true`.

File: schema/artist/highlights.js [hunk 3]
```diff
@@ -1,23 +1,51 @@
 import { GraphQLBoolean, GraphQLObjectType, GraphQLList, GraphQLString } from "graphql"
```
Comment (mzikherman) [inline schema/artist/highlights.js]: If all of the fields for market data are just going to be root on the artist, we could just remove this file completely.

Or, we can duplicate a bit: allow partners to exist on artist, but also have this (maybe more specialized/filtered) call here.

File: schema/partner_artist.js [hunk 4]
```diff
@@ -61,3 +69,12 @@ const PartnerArtist = {
 }
 
 export default PartnerArtist
+
+// The below can be used as the connection from an artist to its partners.
+// The edge is the PartnerArtist relationship, with the node being the partner.
+export const PartnerArtistConnection = connectionDefinitions({
```
Comment (mzikherman) [inline schema/partner_artist.js]: This connection definition was the key part. You can customize the edge/node type and fields for a connection...but then you may need to do some mucking when you're munging your API response into the form the connection expects.

File: schema/partner_artist.js [hunk 5]
```diff
@@ -4,33 +4,36 @@ import Artist from "./artist/index"
 import numeral from "./fields/numeral"
 import { IDFields } from "./object_identification"
 import { GraphQLString, GraphQLObjectType, GraphQLNonNull, GraphQLBoolean } from "graphql"
+import { connectionDefinitions } from "graphql-relay"
 
-export const PartnerArtistType = new GraphQLObjectType({
```
Comment (mzikherman) [inline schema/partner_artist.js]: I flattened out some of the fields here b/c I needed to access them in the `connectionDefinitions` below. I probably could have somehow kept them as-is and still been able to access/use them, but I couldn't figure it out, so I just started refactoring/flattening out. The end result is identical, but I'm able to just use `fields` below and it worked as expected.

File: schema/artist/highlights.js [hunk 6]
```diff
@@ -1,23 +1,51 @@
...
+              {},
+              edge
+            )
+            newEdge.node = edge.node.partner
+            newEdges = newEdges.concat(newEdge)
+          })
+          connection.edges = newEdges
+
+          return connection
+        })
```
Comment (alloy) [inline schema/artist/highlights.js]: Is this identical to the one below? If so, maybe consolidate that in a single exported function?

Comment (mzikherman) [reply schema/artist/highlights.js]: Yep! Was just testing/playing around.

**Comment by mzikherman** [comment]
I think this might be ready for a merge?!

Also once it's in I'll update the schema for market insights - there'll be some minor code updates needed there.

**Comment by alloy** [comment]
Yep, this LGTM 👍

Can you please soon-ish replace the data mucking with a forked version of the utility function, though? This current code isn’t a great example of what rich connections should look like.

**Pull request merged**